app/models/asset.rb

```diff
@@ -109,6 +109,12 @@ def self.all_derivative_count
     ).first["count"]
   end
 
+  def self.all_restricted_derivative_count
+    Kithe::Asset.connection.select_all(
+      "SELECT count(*) FROM (SELECT row_to_json(jsonb_each(file_data -> 'derivatives')) -> 'value' ->> 'storage' as storage_key FROM kithe_models WHERE kithe_model_type = 2 ) as derivatives where storage_key = 'restricted_kithe_derivatives'"
+    ).first['count']
+  end
+
```

Comment:
I'm not totally sure if I like this being in Asset model. 

I realize `self.all_derivative_count` is in model.... this new one is a SUBSET of that one, rather than alternate number? It's getting a bit confusing. 

One of the places we use all_derivative_count is for a total for any progress bar taht's operating on all derivatives.... although that's mostly just weird edge case ones at present.  Some of those might be wrong now, cause the total number will be ALL derivatives, but what they DO might just be on public derivatives (so may never get to 100%). 

This thing of putting restricted derivs on a different bucket is a bit convoluted, alas. :(

Not sure where toher than the orphan checker we're going to use this one JUST for restricted derivs... what if we just inline this were we need it?  Or alternately, just put something in a comment making it clear this is a SUBSET of above one. It probably doens't matter too much where it is.

Reply:
I'd honestly rather move it to the rake task. I doubt we will ever use this query anywhere else.

Reply:
Done